instrumentation/nocode/src/main/java/com/splunk/opentelemetry/instrumentation/nocode/YamlParser.java

```diff
@@ -19,93 +19,233 @@
+        NocodeExpression spanName = toExpression(yamlRule.get("spanName"));
+        SpanKind spanKind = null;
+        if (yamlRule.get("spanKind") != null) {
+          String spanKindString = yamlRule.get("spanKind").toString();
+          try {
+            spanKind = SpanKind.valueOf(spanKindString.toUpperCase(Locale.ROOT));
+          } catch (IllegalArgumentException exception) {
+            logger.warning("Invalid span kind " + spanKindString);
           }
-          NocodeExpression spanStatus = toExpression(yamlRule.get("spanStatus"));
-
-          Map<String, NocodeExpression> ruleAttributes = new HashMap<>();
-          List<Map<String, Object>> attrs = (List<Map<String, Object>>) yamlRule.get("attributes");
-          if (attrs != null) {
-            for (Map<String, Object> attr : attrs) {
-              ruleAttributes.put(attr.get("key").toString(), toExpression(attr.get("value")));
-            }
+        }
+        NocodeExpression spanStatus = toExpression(yamlRule.get("spanStatus"));
+
+        Map<String, NocodeExpression> ruleAttributes = new HashMap<>();
+        List<Map<String, Object>> attrs = (List<Map<String, Object>>) yamlRule.get("attributes");
+        if (attrs != null) {
+          for (Map<String, Object> attr : attrs) {
+            ruleAttributes.put(attr.get("key").toString(), toExpression(attr.get("value")));
           }
-          answer.add(
-              new NocodeRules.Rule(
-                  className, methodName, spanName, spanKind, spanStatus, ruleAttributes));
         }
+        answer.add(
+            new NocodeRules.Rule(
+                classMatcher, methodMatcher, spanName, spanKind, spanStatus, ruleAttributes));
       }
     }
 
     return answer;
   }
 
+  private ElementMatcher<TypeDescription> classMatcher(Object yaml) {
+    if (yaml instanceof String) {
+      return ElementMatchers.named(yaml.toString());
+    }
+    return matcherFromYaml(yaml, makeParsers(MatcherParser.hasSuperType));
+  }
+
+  private ElementMatcher<MethodDescription> methodMatcher(Object yaml) {
+    if (yaml instanceof String) {
+      return ElementMatchers.named(yaml.toString());
+    }
+    return matcherFromYaml(
+        yaml, makeParsers(MatcherParser.hasParameterCount, MatcherParser.hasParameterOfType));
+  }
+
+  private static Map<String, MatcherParser> makeParsers(MatcherParser... extras) {
+    Map<String, MatcherParser> answer = new HashMap<>();
+    List<MatcherParser> parsers =
+        new ArrayList<>(
+            Arrays.asList(
+                MatcherParser.and,
+                MatcherParser.or,
+                MatcherParser.not,
+                MatcherParser.named,
+                MatcherParser.nameMatches));
+    parsers.addAll(Arrays.asList(extras));
+    for (MatcherParser mp : parsers) {
+      answer.put(mp.name(), mp);
+    }
+    return answer;
+  }
+
+  private static List<ElementMatcher> matcherListFromYaml(
```

Comment:
you could add a type parameter to `ElementMatcher` to reduce the amount of warnings